src/nvim/memline.c

```diff
@@ -1441,17 +1441,17 @@ recover_names (
  * Append the full path to name with path separators made into percent
  * signs, to dir. An unnamed buffer is handled as "" (<currentdir>/"")
  */
-static char_u *make_percent_swname(char_u *dir, char_u *name)
+static char *make_percent_swname(char *dir, char *name)
```

Comment:
- `dir`can be const.
- `dir` cannot be NULL.
- `name` can be const, but requires cascading changes into `fix_fname` and others.
So, this could be 
```c
static char *make_percent_swname(const char *dir, char *name)
  FUNC_ATTR_NONNULL_ARG(1)
```
by now.